projects/dotcms-field-elements/src/components/dot-date-range/dot-date-range.e2e.ts

```diff
@@ -1,147 +1,349 @@
-        await page.waitForChanges();
-        const hint = await element.find('.dot-field__hint');
-        expect(hint).toBeNull();
-    });
+            it('should be invalid, untouched, pristine & required when no option set on load', async () => {
+                await page.setContent(`<dot-date-range name='dateRange' required="true"></dot-date-range>`);
+                element = await page.find('dot-date-range');
+                await page.waitForChanges();
+                expect(element).toHaveClasses(dotTestUtil.class.emptyRequiredPristine);
+            });
 
-    it('it should set value dates in blank, when no valid "value" passed', async () => {
-        element.setProperty('value', { noValid: true });
-        await page.waitForChanges();
-        const dateValue = await element.getProperty('value');
-        expect(dateValue).toBe('');
+            it('should be pristine, untouched & valid when loaded with no options', async () => {
+                await page.setContent(`<dot-date-range name='dateRange'></dot-date-range>`);
+                element = await page.find('dot-date-range');
+                expect(element).toHaveClasses(dotTestUtil.class.empty);
+            });
+        });
     });
 
-    describe('flatpickr interaction', () => {
-        let calendar: E2EElement;
+    describe('@Props', () => {
         beforeEach(async () => {
-            await input.click();
-            await page.waitForChanges();
-            calendar = await page.find('.flatpickr-calendar');
+            page = await newE2EPage();
+            await page.setContent(`<dot-date-range></dot-date-range>`);
+            element = await page.find('dot-date-range');
+            input = await page.find('input');
         });
 
-        it('should open flatpickr when click on input', async() => {
-            expect(calendar.classList.contains('open')).toBe(true);
+        describe('disabled', () => {
+            it('should render attribute', async () => {
+                element.setProperty('disabled', true);
+                await page.waitForChanges();
+                expect(input.getAttribute('disabled')).toBeDefined();
+            });
+
+            it('should not break with invalid data --> truthy', async () => {
+                element.setProperty('disabled', ['a', 'b']);
+                await page.waitForChanges();
+                expect(input.getAttribute('disabled')).toBeDefined();
+            });
+
+            it('should not break with invalid data --> falsy', async () => {
+                element.setProperty('disabled', 0);
+                await page.waitForChanges();
+                expect(input.getAttribute('disabled')).toBeNull();
+            });
         });
 
-        describe('date range set', () => {
-            beforeEach(async () => {
-                const days = await page.findAll('.flatpickr-day');
-                days[5].click();
-                days[8].click();
+        describe('name', () => {
```

Comment:
How do we test [this](https://github.com/dotCMS/core-web/blob/f815f776bfa3f9ad270b2c551ff5878773248eec/projects/dotcms-field-elements/src/components/dot-date-range/dot-date-range.tsx#L141)?

Reply:
no idea how to test that!